fe/fe-core/src/main/java/com/starrocks/sql/optimizer/base/HashDistributionDesc.java

```diff
@@ -39,7 +39,9 @@ public enum SourceType {
         // e.g. required SHUFFLE_JOIN(a, b, c), child SHUFFLE_JOIN(a, b, c), satisfy
         SHUFFLE_JOIN, // hash property from shuffle join
         BUCKET, // hash property from bucket
-        SHUFFLE_ENFORCE // parent node which can not satisfy the requirement will enforce child this hash property
+        SHUFFLE_ENFORCE, // parent node which can not satisfy the requirement will enforce child this hash property
+        SHUFFLE_SET, // hash property from set operator
```

Comment:
I think it's same as `SHUFFLE_JOIN`

Reply:
it seems same as shuffle_join,  however there are many special code involving SHUFFLE_JOIN, so I afraid that set opertor re-using SHUFFLE_JOIN would pollute already existing code.